src/docdb/getCosmosClient.ts

```diff
@@ -4,15 +4,70 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { CosmosClient } from "@azure/cosmos";
+import { getConfiguredAuthProviderId } from "@microsoft/vscode-azext-azureauth";
 import { appendExtensionUserAgent } from "@microsoft/vscode-azext-utils";
 import * as https from "https";
 import * as vscode from 'vscode';
 import { ext } from "../extensionVariables";
 
-export function getCosmosClient(endpoint: string, key: string, isEmulator: boolean | undefined): CosmosClient {
+export type CosmosDBKeyCredential = {
+    type: "key";
+    key: string;
+};
+
+export type CosmosDBAuthCredential = {
+    type: "auth";
+};
+
+export type CosmosDBCredential = CosmosDBKeyCredential | CosmosDBAuthCredential;
+
+/**
+ * Duplicated from @microsoft/vscode-azext-azureauth
+ * @todo: Use the subscription client once it supports customizing scopes.
+ */
+export async function getSessionFromVSCode(scopes: string | string[], options?: vscode.AuthenticationGetSessionOptions): Promise<vscode.AuthenticationSession | undefined> {
+    const scopesArray = typeof scopes === "string" ? [scopes] : scopes;
+    return await vscode.authentication.getSession(getConfiguredAuthProviderId(), scopesArray, options);
```

Comment:
Nit: use the internal auth module:
![image](https://github.com/microsoft/vscode-cosmosdb/assets/21112697/eacabea8-3534-4c4b-850b-85691c4393bc)